### ISSUE TITLE
Process uTP `OFFER` stream

### ISSUE THREAD

**Issue by ogenev** [ethereum/trin#289]
This is related to the way we are processing uTP streams mentioned in https://github.com/ethereum/trin/issues/288.

We want to process and store in DB the received uTP `OFFER` stream here:
https://github.com/ethereum/trin/blob/25ddc6840f07d93611960b47c9e99daaf1887147/trin-core/src/utp/stream.rs#L278

**Comment by ogenev** [comment]
Blocked by https://github.com/ethereum/trin/issues/302.

**Comment by ogenev** [comment]
Superseded by https://github.com/ethereum/trin/issues/313.
